include/llama.h

```diff
@@ -1169,11 +1169,30 @@ extern "C" {
     // NOTE: Used by llama.cpp examples, avoid using in third-party apps. Instead, do your own performance measurements.
     //
 
+    // performance timing information
+    struct llama_perf_data {
+        // llama_context
+        double t_start_ms;
+        double t_load_ms;
+        double t_p_eval_ms;
+        double t_eval_ms;
+
+        int32_t n_p_eval;
+        int32_t n_eval;
+
+        // llama_sampler_chain
+        double t_sample_ms;
+
+        int32_t n_sample;
+    };
+
     enum llama_perf_type {
         LLAMA_PERF_TYPE_CONTEXT       = 0,
         LLAMA_PERF_TYPE_SAMPLER_CHAIN = 1,
     };
 
+    LLAMA_API struct llama_perf_data llama_perf_get(const void * ctx, enum llama_perf_type type);
```

Comment:
I think it would be preferable to have two separate functions, just to remove the possibility of calling it with the wrong type of pointer.